README.md

```diff
@@ -7,14 +7,19 @@
 
 A `SmolStr` is a string type that has the following properties:
 
-  * `size_of::<SmolStr>() == size_of::<String>()`
-  * Strings up to 22 bytes long do not use heap allocations
-  * Runs of `\n` and space symbols (typical whitespace pattern of indentation
-    in programming laguages) do not use heap allocations
-  * `Clone` is `O(1)`
+* `size_of::<SmolStr>() == size_of::<String>()`
+* Strings of up to 22 bytes are stack-allocated. As an optimization, leading newlines
+are compressed as a single byte, as well as consecutive whitespace found directly after
+the leading newlines. For instance, `"\n\n  hello world"` is stored in 13 bytes,
```

Comment:
Ah, unfortunatelly this is not entirely correct: we only compress strings which consist solely of `\n` and ` ` characters. 

To be precise, any substring of [this string](https://github.com/matklad/smol_str/blob/360b36a0e8895e07959d2b65b3a51dbc7c02ab3c/src/lib.rs#L135) will be allocated on the stack.

Reply:
Yes!